src/python/pants/backend/python/dependency_inference/rules_test.py

```diff
@@ -744,6 +744,40 @@ def test_infer_python_strict_multiple_resolves(imports_rule_runner: PythonRuleRu
         )
 
 
+def test_infer_python_identical_files_with_relative_imports_should_be_treated_differently(
+    imports_rule_runner: PythonRuleRunner,
+) -> None:
+    # dependency inference shouldn't cache _just_ based on file contents, because this can break
+    # relative imports. When b reused a's results, b/__init__.py was incorrectly depending on
+    # a/file.py
```

Comment:
If you're making any more changes, can you link the relevant issue (as a "real world" example)